test/AutoDiff/silgen_thunking/main.swift

```diff
@@ -10,6 +10,17 @@
 import StdlibUnittest
 import DifferentiationUnittest
 
+// Verify that SILGen derivative thunks are never `[transparent]`.
+@differentiable(vjp: vjpNoReabstraction)
```

Comment:
This is not testing TBDGen. Please also add a test to tbdgen.swift.

Reply:
Done in https://github.com/apple/swift/pull/27752/commits/f337af96fe48ab0f2e918b2271e7994bcbc3a279.